File: modules/wem-security/src/main/java/com/enonic/wem/security/UserStoreKey.java
package com.enonic.wem.security;

public final class UserStoreKey
{
    private final static UserStoreKey SYSTEM = new UserStoreKey( "system" );

    private final String id;

    public UserStoreKey( final String id )
    {
        this.id = id;
    }

    @Override
    public String toString()
    {
        return this.id;
    }

    @Override
    public boolean equals( final Object o )
    {
        if ( this == o )
        {
            return true;
        }
        if ( !( o instanceof UserStoreKey ) )
        {
            return false;
        }
        final UserStoreKey userStoreKey = (UserStoreKey) o;
        return id.equals( userStoreKey.id );
    }

    @Override
    public int hashCode()
    {
        return id.hashCode();
    }

    public static UserStoreKey system()
    {
        return SYSTEM;
    }
}

File: modules/admin-ui/src/main/resources/web/admin/common/js/security/UserStoreKey.ts
module api.security {

    export class UserStoreKey implements api.Equitable {

        public static SYSTEM: UserStoreKey = new UserStoreKey('system');

        private id: string;

        constructor(id: string) {
            api.util.assert(!api.util.StringHelper.isBlank(id), "UserStoreKey id cannot be null or empty");
            this.id = id;
        }

        isSystem(): boolean {
            return this.id === UserStoreKey.SYSTEM.id;
        }

        toString(): string {
            return this.id;
        }

        getId(): string {
            return this.id;
        }

        static fromString(value: string): UserStoreKey {
            return new UserStoreKey(value);
        }

        equals(o: api.Equitable): boolean {
            if (!api.ObjectHelper.iFrameSafeInstanceOf(o, UserStoreKey)) {
                return false;
            }

            var other = <UserStoreKey>o;
            return this.id === other.id;
        }
    }
}